training-service/src/app/content/subcategory/component/subcategory-name/subcategory-name.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-subcategory-name',
+  templateUrl: './subcategory-name.component.html',
+  styleUrls: ['./subcategory-name.component.scss']
+})
+export class SubcategoryNameComponent implements OnInit {
+
+  name: string | undefined;
+
+  constructor(private route: ActivatedRoute) { }
+
+  ngOnInit() {
+    this.route.params.subscribe(p => this.name = p['name'])
```

Comment:
Unsubscribe is missing for this subscription, this is a potential memory leak.

Reply:
done